extensions/vertx-core/runtime/src/main/java/io/quarkus/vertx/core/runtime/VertxCoreRecorder.java

```diff
@@ -509,18 +486,41 @@ public Integer get() {
         };
     }
 
-    public ThreadFactory createThreadFactory() {
-        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+    public ThreadFactory createThreadFactory(LaunchMode launchMode) {
+        Optional<ClassLoader> nonDevModeTccl;
+        if (launchMode == LaunchMode.DEVELOPMENT) {
+            currentDevModeNewThreadCreationClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
nitpicking: `currentDevModeNewThreadCreationClassLoader` says:
> This must be accessed under the {@link #devModeThreads} lock

Yet, I don't see a lock/syncronized here, probably because it's not needed because there is no concurrency here.

Reply:
I was looking at the same :) And concluded that it's fine, but yea we could make it synchronized to make it more explicit.

Reply:
... or adjust the javadoc or leave an inline comment why it's not needed here (which brings us back to my other suggestion to extract a common method).